test/e2e/helpers.go

```diff
@@ -911,36 +894,16 @@ func WaitForDaemonset(ctx context.Context, input clusterctl.ApplyClusterTemplate
 	}, input.WaitForControlPlaneIntervals...).Should(Equal(true))
 }
 
-// podListHasNumPods fulfills the cluster-api PodListCondition type spec
-// given a list of pods, we validate for an exact number of those pods in a Running state
-func podListHasNumPods(numPods int) func(pl *corev1.PodList) error {
-	return func(pl *corev1.PodList) error {
-		var runningPods int
-		for _, p := range pl.Items {
-			if p.Status.Phase == corev1.PodRunning {
-				runningPods++
-			}
-		}
-		if runningPods != numPods {
-			return errors.Errorf("expected %d Running pods, got %d", numPods, runningPods)
-		}
-		return nil
-	}
-}
-
-// podListHasAtLeastNumPods fulfills the cluster-api PodListCondition type spec
-// given a list of pods, we validate for at least 1 number of those pods in a Running state
-func podListHasAtLeastNumPods(numPods int) func(pl *corev1.PodList) error {
-	return func(pl *corev1.PodList) error {
-		var runningPods int
-		for _, p := range pl.Items {
-			if p.Status.Phase == corev1.PodRunning {
-				runningPods++
-			}
+// WaitForDeployment retries during E2E until a deployments's pods are all Running.
+func WaitForDeployment(ctx context.Context, input clusterctl.ApplyClusterTemplateAndWaitInput, cl client.Client, name, namespace string) {
```

Comment:
we already have 

```
// WaitForDeploymentsAvailable waits until the Deployment has status.Available = True, that signals that
// all the desired replicas are in place.
// This can be used to check if Cluster API controllers installed in the management cluster are working.
func WaitForDeploymentsAvailable(ctx context.Context, input WaitForDeploymentsAvailableInput, intervals ...interface{})
```

could it be used instead of adding this new helper? It essentially does the same thing as far as I can tell